shell/shell.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2022 Jetpack Technologies Inc and contributors. All rights reserved.
+// Use of this source code is governed by the license in the LICENSE file.
+
+// Package shell detects the user's default shell and configures it to run in
+// Devbox.
+package shell
```

Comment:
Totally cool with this package, but just to mention it, another option could be to place `shell.go` inside `nix`. It could make the nix package self-contained and potentially re-usable by others.

You can leave as is too if you prefer.

Reply:
I think I like that idea. I'll try combining the two and see how it works out.

Reply:
I'll merge this first so we have something working and then see about merging the packages in a followup.

Reply:
@loreto I merged them in #93 and I think it works really well.